### ISSUE TITLE
set paste is crush with jedi

### ISSUE THREAD

**Issue by becxer** [davidhalter/jedi-vim#484]
'set paste' in .vimrc is crush with jedi

When I removed 'set paste' jedi works well

**Comment by blueyed** [comment]
Can you be more descriptive, please?

What is the problem?  Are completions not showing up?

Apart from that, why would you have `set paste` in your default config?

**Comment by davidhalter** [comment]
@blueyed While I agree with your questions, we should actually think about this a bit more:

1. He's not the first one with this problem, he's at least the fourth.
2. Maybe there's ways around this, so that Jedi still works - or we could document it better.

**Comment by becxer** [comment]
Sorry for less information , cause I'm not good at english.

The problem is Jedi-vim python auto-complete no working when I 'set paste' in .vimrc

I usually used set paste with default setting because I need to copy paste many time when I work in company.

Maybe quick solution is remove set paste or toggle paste with shortcut, but I think there are some people like me using paste option default. 

And if you need, I want to fix it for fork of this project

**Comment by davidhalter** [comment]
Toggling is actually what you should do, IMO.

**Comment by blueyed** [comment]
> think there are some people like me using paste option default

That means you're missing out a lot of Vim's features, linke autoindenting etc.

I was not aware that this is a problem with jedi-vim, and agree that it should be fixed.
I still need to investigate / try it, because it's still unclear to me what is not working, but that might be obvious then.

**Comment by blueyed** [comment]
@becxer 
> And if you need, I want to fix it for fork of this project

A fix for this through a PR would be great, of course - but I can imagine that it's not trivial.

**Comment by davidhalter** [comment]
I'm not even sure if it's possible. I would just say that in that case we should at least improve documentation.

**Comment by blueyed** [comment]
I've looked into this now, and the point of `:set paste` is explicitly to disable insert mode mappings etc.  See http://vimdoc.sourceforge.net/htmldoc/options.html#%27paste%27.

I've noticed however, that using `<c-space>` to trigger a completion would end insert mode and enter the text from the previous insert - but that also happens without jedi-vim being loaded, and even `vim -u NONE`.

So, after all, while we might add something to the documentation I think `'paste'` is a basic concept of Vim, and it's expected that it does not work.

@becxer 
It would be interesting to know what is happening for you.  Is it that just the completion does not work, or were you affected also by the `<c-space>` behaviour?  Something else?
